### PR TITLE
Add sidechain additional fee in outputpayload

### PR DESCRIPTION
**Pull request by xiaowenwe**

### CODE REVIEW

File: core/types/outputpayload/transfercrosschainfee.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"errors"
+	"io"
+
+	"github.com/elastos/Elastos.ELA/common"
+	"github.com/elastos/Elastos.ELA/crypto"
+)
+
+type CrossChainFee struct {
```
Comment (alexpan0610) [inline core/types/outputpayload/transfercrosschainfee.go]: AdditionalFee is better than CrossChainFee, and the filename should as the same as the class name.
Add comments to your code to explain why this output payload exists, and what's the parameters use for.
See `core/types/outputpayload/mapping.go` for example

File: core/types/outputpayload/transfercrosschainfee.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	Version byte
+	TxHash  []byte
```
Comment (alexpan0610) [inline core/types/outputpayload/transfercrosschainfee.go]: In out project, a hash is always a Uint256 type, the TxHash should be a common.Uint256 not a []byte.

File: core/types/outputpayload/transfercrosschainfee_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package outputpayload
+
+import (
+	"bytes"
+	"github.com/elastos/Elastos.ELA/common"
```
Comment (alexpan0610) [inline core/types/outputpayload/transfercrosschainfee_test.go]: Attention to the import format, it's hidden in the goland IDE.

File: core/types/outputpayload/transfercrosschainfee.go (hunk 1 above)
Comment (xiaowenwe) [reply core/types/outputpayload/transfercrosschainfee.go]: Already changed as required

File: core/types/outputpayload/transfercrosschainfee.go (hunk 2 above)
Comment (xiaowenwe) [reply core/types/outputpayload/transfercrosschainfee.go]: Already changed as required

File: core/types/outputpayload/transfercrosschainfee_test.go (hunk 3 above)
Comment (xiaowenwe) [reply core/types/outputpayload/transfercrosschainfee_test.go]: Already changed as required

File: core/types/outputpayload/additionalfee_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+	buf := new(bytes.Buffer)
+	if err := ccf1.Serialize(buf); err != nil {
+		t.Error("additional fee serialize failed")
+	}
+	if !bytes.Equal(buf.Bytes(), cBytes) {
+		t.Error("additional fee serialize failed\n", common.BytesToHexString(buf.Bytes()))
+	}
+}
+func TestAdditionalFee_Deserialize(t *testing.T) {
```
Comment (alexpan0610) [inline core/types/outputpayload/additionalfee_test.go]: Add blank line above.

File: core/types/outputpayload/additionalfee_test.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		t.Error("additional fee deserialize failed")
+	}
+	if ccf.Version != 0 {
+		t.Error("error version")
+	}
+	if common.BytesToHexString(ccf.TxHash.Bytes()) != hash1 {
+		t.Error("error hash")
+	}
+}
+func TestAdditionalFee_Validate(t *testing.T) {
```
Comment (alexpan0610) [inline core/types/outputpayload/additionalfee_test.go]: Add blank line above.

**Comment by xiaowenwe** [comment]
No need for this feature

**Pull request closed without merging**